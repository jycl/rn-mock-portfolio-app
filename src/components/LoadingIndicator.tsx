import React, { useRef } from 'react';
import { Animated, Easing, StyleSheet } from 'react-native';
import Assets from '../assets';

/**
 * BitCoin logo loading indicator displayed when screen is loading.
 */
export default function LoadingIndicator() {
  const rotationDeg = useRef(new Animated.Value(0)).current;
  Animated.loop(
    Animated.timing(rotationDeg, {
      toValue: 1,
      duration: 5000,
      easing: Easing.linear,
      useNativeDriver: true,
    }),
  ).start();

  const spin = rotationDeg.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg'],
  });
  return (
    <Animated.Image
      style={[styles.image, { transform: [{ rotate: spin }] }]}
      source={Assets.btc}
    />
  );
}

const styles = StyleSheet.create({
  image: {
    width: 80,
    height: 80,
  },
});
