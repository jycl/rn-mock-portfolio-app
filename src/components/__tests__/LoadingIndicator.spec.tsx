import React from 'react';
import TestRenderer from 'react-test-renderer';
import LoadingIndicator from '../LoadingIndicator';

describe('LoadingIndicator component', () => {
  it('should match snapshot rendering', () => {
    const renderItem = TestRenderer.create(<LoadingIndicator />);
    expect(renderItem.toJSON()).toMatchSnapshot();
  });
});
