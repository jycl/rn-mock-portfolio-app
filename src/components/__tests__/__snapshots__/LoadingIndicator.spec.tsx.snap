// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator component should match snapshot rendering 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/btc_logo.png",
    }
  }
  style={
    Object {
      "height": 80,
      "transform": Array [
        Object {
          "rotate": "0deg",
        },
      ],
      "width": 80,
    }
  }
/>
`;
